frontend/src/components/Scopes.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { useEffect } from "react";
+import { Checkbox } from "src/components/ui/checkbox";
+import { Label } from "src/components/ui/label";
+import { cn } from "src/lib/utils";
+import {
+  NIP_47_GET_BALANCE_METHOD,
+  NIP_47_GET_INFO_METHOD,
+  NIP_47_LIST_TRANSACTIONS_METHOD,
+  NIP_47_LOOKUP_INVOICE_METHOD,
+  NIP_47_NOTIFICATIONS_PERMISSION,
+  NIP_47_PAY_INVOICE_METHOD,
+  ReadOnlyScope,
+  SCOPE_GROUP_CUSTOM,
+  SCOPE_GROUP_FULL_ACCESS,
+  SCOPE_GROUP_READ_ONLY,
+  Scope,
+  ScopeGroup,
+  WalletCapabilities,
+  scopeDescriptions,
+  scopeGroupDescriptions,
+  scopeGroupIconMap,
+  scopeGroupTitle,
+} from "src/types";
+
+interface ScopesProps {
+  capabilities: WalletCapabilities;
+  scopes: Set<Scope>;
+  onScopeChange: (scopes: Set<Scope>) => void;
+}
+
+const isSetEqual = (setA: Set<string>, setB: Set<string>) =>
+  setA.size === setB.size && [...setA].every((value) => setB.has(value));
+
+const Scopes: React.FC<ScopesProps> = ({
+  capabilities,
+  scopes,
+  onScopeChange,
+}) => {
+  const fullAccessScopes: Set<Scope> = React.useMemo(() => {
+    return new Set(capabilities.scopes);
+  }, [capabilities.scopes]);
+
+  const readOnlyScopes: Set<ReadOnlyScope> = React.useMemo(() => {
+    const scopes: Scope[] = capabilities.scopes;
+    return new Set(
+      scopes.filter((method): method is ReadOnlyScope =>
+        [
+          NIP_47_GET_BALANCE_METHOD,
+          NIP_47_GET_INFO_METHOD,
+          NIP_47_LOOKUP_INVOICE_METHOD,
+          NIP_47_LIST_TRANSACTIONS_METHOD,
+          NIP_47_NOTIFICATIONS_PERMISSION,
+        ].includes(method)
+      )
+    );
+  }, [capabilities.scopes]);
+
+  const [scopeGroup, setScopeGroup] = React.useState<ScopeGroup>(() => {
+    if (!scopes.size || isSetEqual(scopes, fullAccessScopes)) {
```

Comment:
We don't need to do this, if an app passes in any request methods it can just be the custom option